internal/bminventory/inventory.go

```diff
@@ -672,19 +632,21 @@ func (b *bareMetalInventory) InstallCluster(ctx context.Context, params installe
 	if err = b.db.Preload("Hosts", "status <> ?", host.HostStatusDisabled).First(&cluster, "id = ?", params.ClusterID).Error; err != nil {
 		return common.NewApiError(http.StatusNotFound, err)
 	}
-	if err = b.refreshAllHosts(ctx, &cluster); err != nil {
+	if _, err = b.clusterApi.RefreshStatus(ctx, &cluster, b.db); err != nil {
 		return common.GenerateErrorResponder(err)
 	}
 
 	// Reload again after refresh
 	if err = b.db.Preload("Hosts", "status <> ?", host.HostStatusDisabled).First(&cluster, "id = ?", params.ClusterID).Error; err != nil {
 		return common.NewApiError(http.StatusNotFound, err)
 	}
-	if err = b.verifyClusterNetworkConfig(ctx, &cluster); err != nil {
-		return common.GenerateErrorResponder(err)
+	// Verify cluster status
+	if swag.StringValue(cluster.Status) != models.ClusterStatusReady {
+		log.WithError(err).Errorf("Cluster %s is not ready to install", params.ClusterID)
+		return common.NewApiError(http.StatusBadRequest, errors.New("Cluster is not ready to install"))
 	}
 
-	if err = b.validateAllHostsCanBeInstalled(&cluster); err != nil {
+	if err = b.verifyClusterNetworkConfig(ctx, &cluster); err != nil {
```

Comment:
If there was already RefreshStatus to the cluster, this is not needed because RefreshStatus includes everything in  verifyClusterNetworkConfig.

Reply:
fixed